### PR TITLE
[TACHYON-1581] Modify the shell of 'thriftGen' to generate private member

### PR DESCRIPTION
**Pull request by ConeyLiu**
Modify the shell of 'thriftGen' to generate private member.
https://tachyon.atlassian.net/projects/TACHYON/issues/TACHYON-1581?filter=allopenissues

### CODE REVIEW

**Comment by apc999** [comment]
Jenkins, test this please.

**Comment by apc999** [comment]
@ConeyLiu  Thanks for the patch, I certainly appreciate the value. Before we go to further code review, I have a high level question: is this change (from `public` to `private`) achieved by editing `common/src/main/java/tachyon/thrift/FileInfo.java`? This file is auto-generated by thrift framework, derived from `common/src/thrift/file_system_master.thrift:17`. We typically don't modify those files as the change will be overwritten easily next time when we update the thrift definition files.

In addition, invoking a method `isIsCompleted` looks strange.

An alternative way to achieve the same goal can be like this: you can change `common/src/thrift/file_system_master.thrift:25` to make the field name from `isCompleted` to `completed`, and then run `bin/tachyon thriftGen`, then you will have `isCompleted()` as a public method available in `common/src/main/java/tachyon/thrift/FileInfo.java`. You can do similar trick to other field like `isPersisted` and etc

**Comment by ConeyLiu** [comment]
@apc999 Thanks for you advise, I will try it.

**Comment by ConeyLiu** [comment]
@apc999 I modify the shell in bin/tachyon as follow:
 "thriftGen")
    rm -rf ${bin}/../common/src/main/java/tachyon/thrift
    for src_file in $(ls ${bin}/../common/src/thrift); do
 -    thrift --gen java -out ${bin}/../common/src/main/java/. ${bin}/../common/src/thrift/${src_file}
 +    thrift --gen java:private-members -out ${bin}/../common/src/main/java/. ${bin}/../common/src/thrift/${src_file}
    done
    exit 0
  ;;

This way, the change of *.thrift will not overwritten. It will generate private member.
Please verify it.

**Comment by apc999** [comment]
@ConeyLiu Thanks for finding the thrift trick. It looks like a good solution. 
One note about the PR: currently, we are merging a few huge PRs to the master branch (e.g., https://github.com/amplab/tachyon/pull/2525 and https://github.com/amplab/tachyon/pull/2523). My suggestion is sync your branch with master branch after these two PRs are merged. After that, you can update this PR again. Does that sound good to you?

**Comment by ConeyLiu** [comment]
@apc999  I fix the conflict. Rerun the `bin/tachyon thriftGen`, and replace use member directly to the function of getter or setter. Please verify it.

**Comment by ConeyLiu** [comment]
Jenkins, test this please.

**Comment by apc999** [comment]
@ConeyLiu  thanks for the PR. the current PR looks good to me. Could you resolve the conflicts?

**Comment by ConeyLiu** [comment]
@apc999 I have resolve the conflicts.Please verify it. Thanks very much.

**Comment by apc999** [comment]
+1 LGTM, thanks @ConeyLiu for he improvement!

File: common/src/thrift/file_system_master.thrift [hunk 1]
```diff
@@ -28,11 +28,11 @@ struct FileInfo {
   5: i64 length
   6: i64 blockSizeBytes
   7: i64 creationTimeMs
-  8: bool isCompleted
-  9: bool isFolder
-  10: bool isPinned
-  11: bool isCacheable
-  12: bool isPersisted
+  8: bool completed
```
Comment (jsimsa) [inline common/src/thrift/file_system_master.thrift]: great!

File: tests/src/test/java/tachyon/worker/block/meta/TieredStoreIntegrationTest.java [hunk 2]
```diff
@@ -133,7 +133,6 @@ public void pinFileTest() throws IOException, TachyonException {
 
     // Confirm the pin with master
     Assert.assertTrue(mTFS.getStatus(file).isPinned());
-
```
Comment (jsimsa) [inline tests/src/test/java/tachyon/worker/block/meta/TieredStoreIntegrationTest.java]: nit: why are we removing these?

**Comment by jsimsa** [comment]
@ConeyLiu thank you for the PR, i really like this improvement ... left one small comment for you about removing blank lines

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
OK to test

**Comment by ConeyLiu** [comment]
Ok. Thanks very much.

Date: Tue, 26 Jan 2016 08:35:01 -0800

**Pull request merged**